### ISSUE TITLE
Alloydb Managed Service issue with oracle_fdw

### ISSUE THREAD

**Issue by imrankhan1983** [laurenz/oracle_fdw#726]
I am trying to get tables from Oracle ExaCC database 19c version from Google alloydb managed service but not VM using Oracle data wrapper extension . I was able to create oracle_fdw in managed service and also db connection and user mapping with postgres user. Also, created foreign table but when I try to query the foreign table it throws "Getting error postgresql error: connection for foreign table cannot be established". Since, this is managed service unable to do telnet and use a utility like psql to try connection to Oracle. Any help on this please.

**Comment by imrankhan1983** [comment]
Oracle_diag() gives 21 oracle client . But the source is oracle 19c ExaCC DB.

**Comment by laurenz** [comment]
Don't use Oracle client version 21 if you have CLOBs in your Oracle database; it won't work.

But your reported problem is a different one. You used an Oracle connect string in the `CREATE SERVER` command that does not work. Perhaps you made a mistake, perhaps the Oracle server cannot be reached from the PostgreSQL server. There is no way I can help you, since you don't give me any information.

**Comment by laurenz** [comment]
Yes, I require more information. Please start with the exact `CREATE SERVER` statement and the exact error message.

**Comment by imrankhan1983** [comment]
> Yes, I require more information. Please start with the exact `CREATE SERVER` statement and the exact error message.

Create extension IF NOT EXISTS oracle_fdw CASCADE;
CREATE SERVER ora_serv FOREIGN DATA WRAPPER  oracle_fdw OPTIONS (dbserver 'myhost.domain:port/service_name');
GRANT USAGE ON FOREIGN SERVER ora_serv TO postgres;
CREATE USER MAPPING FOR postgres SERVER ora_serv OPTIONS ( USER 'ORA_SCHEMA', PASSWORD 'password');
CREATE SCHEMA new_schema; 
Tried first one table:
CREATE FOREIGN TABLE new_schema.table1 ( code_name varchar, sub_name varchar, value numeric) server ora_serv OPTIONS (TABLE 'ORA_TABLE1');

**Comment by imrankhan1983** [comment]
Those are the steps I followed. Please let me know if you require more info.

**Comment by laurenz** [comment]
Try starting the connect string with two slashes.
Please copy and paste the complete error message.
Perhaps you cannot reach `myhost.domain` from the PostgreSQL server? You might want to ask your hosting provider.

**Comment by imrankhan1983** [comment]
postgresql error: connection for foreign table 'ora_table1' cannot be established

**Comment by laurenz** [comment]
I need the detail message from the error message too. That contains the Oracle error that might give you a clue.

**Comment by imrankhan1983** [comment]
> I need the detail message from the error message too. That contains the Oracle error that might give you a clue.

Unable to see any in alloydb logs

**Comment by laurenz** [comment]
I am sorry, but I cannot help you without information. Try to contact alloydb, perhaps they can provide the necessary information.

**Comment by imrankhan1983** [comment]
> I am sorry, but I cannot help you without information. Try to contact alloydb, perhaps they can provide the necessary information.

Thank you so much for your support. I raised a case with alloydb support . Will update you soon

**Comment by laurenz** [comment]
Do you have any new information?
Do we need to keep the issue open?

**Comment by imrankhan1983** [comment]
Hi Laurenz/Oracle_fdw,

 Thank you so much for your humble follow up. I am sorry for not updating.
Seems like it's issue with GCP private service with our vpc . So, we are
not going with oracle_fdw. This can be closed.

Thanks,
Imran
